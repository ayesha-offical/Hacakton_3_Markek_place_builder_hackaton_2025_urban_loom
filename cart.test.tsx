import { render, screen, waitFor } from "@testing-library/react";
import Cart from "@/components/cart";
import useCartStore from "./store";
import "@testing-library/jest-dom";

jest.mock("./store", () => ({
  __esModule: true,
  default: jest.fn(),
}));

const setFakeStore = (groupedItems: any[]) => {
  (useCartStore as unknown as jest.Mock).mockImplementation((selector) =>
    selector({ getGroupedItems: () => groupedItems })
  );
};

describe("Cart Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("renders correctly with 0 items when groupedItems returns empty array", async () => {
    setFakeStore([]);
    const { container } = render(<Cart />);
    
    await waitFor(() => {
      expect(screen.getByText(/cart/i)).toBeInTheDocument();
    });
    
    // Updated to check for separate elements
    expect(screen.getByText("0")).toBeInTheDocument();
    expect(screen.getByText("items")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("renders correctly with correct item count when groupedItems returns items", async () => {
    setFakeStore([{ id: 1 }, { id: 2 }, { id: 3 }]);
    const { container } = render(<Cart />);
    
    await waitFor(() => {
      expect(screen.getByText(/cart/i)).toBeInTheDocument();
    });
    
    // Updated to check for separate elements
    expect(screen.getByText("3")).toBeInTheDocument();
    expect(screen.getByText("items")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("renders the link with correct href attribute", async () => {
    setFakeStore([]);
    render(<Cart />);
    
    await waitFor(() => {
      const link = screen.getByRole("link");
      expect(link).toHaveAttribute("href", "/cart");
    });
  });
});