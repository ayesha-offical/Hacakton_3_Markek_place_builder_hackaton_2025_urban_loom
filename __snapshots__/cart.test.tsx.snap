// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component renders correctly with 0 items when groupedItems returns empty array 1`] = `
<div>
  <a
    class="flex items-center md:text-sm gap-2 border border-gray-200 px-2 py-1 rounded-md shadow-md hover:shadow-none text-xs "
    href="/cart"
  >
    <svg
      class="lucide lucide-shopping-cart text-cyan-950 w-5 h-5 md:w-6 md:h-6 hoverEffect"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="21"
        r="1"
      />
      <circle
        cx="19"
        cy="21"
        r="1"
      />
      <path
        d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
      />
    </svg>
    <div
      class="flex flex-col "
    >
      <p
        class="text-xs"
      >
        <span
          class="font-semibold "
        >
          0
        </span>
        items
      </p>
      <p
        class="font-semibold md:text-base text-xs"
      >
        Cart
      </p>
    </div>
  </a>
</div>
`;

exports[`Cart Component renders correctly with correct item count when groupedItems returns items 1`] = `
<div>
  <a
    class="flex items-center md:text-sm gap-2 border border-gray-200 px-2 py-1 rounded-md shadow-md hover:shadow-none text-xs "
    href="/cart"
  >
    <svg
      class="lucide lucide-shopping-cart text-cyan-950 w-5 h-5 md:w-6 md:h-6 hoverEffect"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="21"
        r="1"
      />
      <circle
        cx="19"
        cy="21"
        r="1"
      />
      <path
        d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
      />
    </svg>
    <div
      class="flex flex-col "
    >
      <p
        class="text-xs"
      >
        <span
          class="font-semibold "
        >
          3
        </span>
        items
      </p>
      <p
        class="font-semibold md:text-base text-xs"
      >
        Cart
      </p>
    </div>
  </a>
</div>
`;
